wasm_resources/index.html

```diff
@@ -0,0 +1,143 @@
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <style>
+      body {
+        margin: 0;
+        background-color: black;
+      }
+      canvas {
+        background-color: black;
+        /*
+          This is a workaround for high DPI screens which winit or something
+          has an issue with where it will resize the canvas to smaller than
+          the full width of the screen.
+        */
+        width: 100vw !important;
+        height: 100vh !important;
+      }
+    </style>
+    <!-- Nifty spinner from: https://github.com/loadingio/css-spinner/blob/088aefa790d015e9c7df664d426374952e65124a/dist/dual-ring.html#L1 -->
+    <style type="text/css">
+      .lds-dual-ring {
+        width: 80px;
+        height: 80px;
+      }
+      .lds-dual-ring:after {
+        content: " ";
+        display: block;
+        width: 64px;
+        height: 64px;
+        border-radius: 50%;
+        border: 6px solid #fff;
+        border-color: #fff transparent #fff transparent;
+        animation: lds-dual-ring 1.2s linear infinite;
+      }
+      @keyframes lds-dual-ring {
+        0% {
+          transform: rotate(0deg);
+        }
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+      .loading-container {
+        z-index: -1;
+        display: flex;
+        flex-direction: column;
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        color: white;
+      }
+
+      .loading-container > div {
+        margin: auto;
+        display: flex;
+        flex-direction: column;
+        color: white;
+        align-items: center;
+      }
+
+      .loading-text {
+        font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
+        margin-top: 2em;
+        font-size: 1.2em;
+      }
+    </style>
+  </head>
+  <body>
+    <div class="loading-container">
+      <div>
+        <div class="lds-dual-ring"></div>
+        <div class="loading-text">Loading game...</div>
+      </div>
+    </div>
+    <script>
+      // Insert hack to make sound autoplay on Chrome as soon as the user interacts with the tab:
+      // https://developers.google.com/web/updates/2018/11/web-audio-autoplay#moving-forward
+      (function () {
+        // An array of all contexts to resume on the page
+        const audioContextList = [];
+
+        // An array of various user interaction events we should listen for
+        const userInputEventNames = [
```

Comment:
I don't really understand what this gets us but I'm assuming adding these input event listeners could be generally useful in the future?

Reply:
All of that is actually solely to fix audio contexts in Chrome and does nothing to impact the game otherwise.

On chrome, the web page isn't allowed to auto-play sounds before the user clicks on the screen or performs some other interaction with the page. what that code does is wait for some interaction and once it gets it, it resumes the audio context, which Chrome will allow because there has been a user interaction at that point.

Reply:
Okay seems fine :+1: